kedro/extras/datasets/tensorflow/tensorflow_model_dataset.py

```diff
@@ -0,0 +1,191 @@
+    DataSetError,
+    Version,
+    get_filepath_str,
+    get_protocol_and_path,
+)
+
+
+class TensorFlowModelDataset(AbstractVersionedDataSet):
+    """``TensorflowModelDataset`` loads and saves TensorFlow models.
+    The underlying functionality is supported by, and passes input arguments through to,
+    TensorFlow 2.X load_model and save_model methods.
+
+    Example:
+    ::
+
+        >>> from kedro.extras.datasets.tensorflow import TensorFlowModelDataset
+        >>> import tensorflow as tf
+        >>> import numpy as np
+        >>>
+        >>> data_set = TensorFlowModelDataset("saved_model_path")
+        >>> model = tf.keras.Model()
+        >>> predictions = model.predict([...])
+        >>>
+        >>> data_set.save(model)
+        >>> loaded_model = data_set.load()
+        >>> new_predictions = loaded_model.predict([...])
+        >>> np.testing.assert_allclose(predictions, new_predictions, rtol=1e-6, atol=1e-6)
+
+    """
+
+    DEFAULT_LOAD_ARGS = {}  # type: Dict[str, Any]
+    DEFAULT_SAVE_ARGS = {"save_format": "tf"}  # type: Dict[str, Any]
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+        version: Version = None,
+        credentials: Dict[str, Any] = None,
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``TensorFlowModelDataset``.
+
+        Args:
+            filepath: Filepath to a TensorFlow model directory prefixed with a protocol
+                like `s3://`. If prefix is not provided `file` protocol (local filesystem)
+                will be used. The prefix should be any protocol supported by ``fsspec``.
+                Note: `http(s)` doesn't support versioning.
+            load_args: TensorFlow options for loading models.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/load_model
+                All defaults are preserved.
+            save_args: TensorFlow options for saving models.
+                Here you can find all available arguments:
+                https://www.tensorflow.org/api_docs/python/tf/keras/models/save_model
+                All defaults are preserved, except for "save_format", which is set to "tf".
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{'token': None}`.
+            fs_args: Extra arguments to pass into underlying filesystem class constructor
+                (e.g. `{"project": "my-project"}` for ``GCSFileSystem``).
```

Comment:
i think you need `open_args_load` and stuff here :)

Reply:
`open_args_load` as well as `open_args_save` are not required since we don't use `self._fs.open()` here.

Reply:
ah of course 🤦‍♂️ my bad